### PR TITLE
Ensure metadata api token header on metadata fetch

### PR DESCRIPTION
**Pull request by xmppjingle**

### CODE REVIEW

**Review by motobob — CHANGES_REQUESTED** [review]
need to handle non EC2 cases.

File: src/erlcloud_ec2_meta.erl [hunk 1]
```diff
@@ -32,7 +45,9 @@ get_instance_metadata(Config) ->
 %%
 get_instance_metadata(ItemPath, Config) ->
     MetaDataPath = "http://" ++ ec2_meta_host_port() ++ "/latest/meta-data/" ++ ItemPath,
-    erlcloud_aws:http_body(erlcloud_httpc:request(MetaDataPath, get, [], <<>>, erlcloud_aws:get_timeout(Config), Config)).
+    {ok, Token} = get_metadata_api_token(Config),
+    Headers = [{"X-aws-ec2-metadata-token", Token}],
+    erlcloud_aws:http_body(erlcloud_httpc:request(MetaDataPath, get, Headers, <<>>, erlcloud_aws:get_timeout(Config), Config)).
```
[the remark this answers is not in the document]
Comment (carlisom) [reply src/erlcloud_ec2_meta.erl:53]: This looks correct, `put` is used to generate the token which is done in `get_metadata_api_token` while the `get` is used to retrieve the meta data. Documentation is [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-metadata-v2-how-it-works.html)

**Review by carlisom — CHANGES_REQUESTED** [review]
Yes, the error from `get_metadata_api_token` needs to be sent to the caller to be handled accordingly vs the current `bad match` that will be encountered.

**Review by carlisom — CHANGES_REQUESTED** [review]
Yes, the error from `get_metadata_api_token` needs to be sent to the caller to be handled accordingly vs the current `bad match` that will be encountered.

**Pull request closed without merging**